k2/csrc/ragged.cu

```diff
@@ -395,43 +414,35 @@ void RaggedShape::Check() {
 
 RaggedShape RaggedShape2(Array1<int32_t> *row_splits, Array1<int32_t> *row_ids,
                          int32_t cached_tot_size) {
-  if (!row_splits && !row_ids) {
-    K2_LOG(FATAL) << "At least one of row_splits and row_ids must be defined";
-  }
+  K2_CHECK(row_splits != nullptr || row_ids != nullptr)
+      << "At least one of row_splits and row_ids must be defined";
+  ContextPtr ctx = ::GetContext(row_splits, row_ids);
   if (cached_tot_size != -1) {
     if (row_ids != nullptr) K2_CHECK_EQ(cached_tot_size, row_ids->Dim());
-    if (row_splits != nullptr) {  // caution: next check may be slow...
-      const auto &row_splits_ref = *row_splits;
-      K2_CHECK_EQ(cached_tot_size, row_splits_ref[row_splits->Dim() - 1]);
+    if (row_splits != nullptr) {
+      // may be slow as it may copy memory from device to host
```

Comment:
If it's slow, can we use debug check ?

Reply:
Let's keep it as-is for now, just comment here to make we are aware of the underlying memory copy. We can profile the program finally and to see if it leads to great performance issues.
 I was thinking maybe we could cache `Array1.Back()` on CPU as we usually access the last elment (another element we may access frequently is `Array1[0]`)